src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -108,50 +147,144 @@ export class StickyLineCandidateProvider extends Disposable implements IStickyLi
 		if (token.isCancellationRequested) {
 			return;
 		}
 		if (outlineModel.children.size !== 0) {
-			const { stickyOutlineElement, providerID } = StickyOutlineElement.fromOutlineModel(outlineModel, this._model?.outlineProviderId);
-			this._model = new StickyOutlineModel(model.uri, modelVersionId, stickyOutlineElement, providerID);
+			// Update the time it took to find the outline model in the debounce information service
+			const newValue = this._updateDebounceInfo.update(textModel, this._sw!.elapsed());
+			if (this._updateScheduler) {
+				this._updateScheduler.defaultDelay = newValue;
+			}
+			const { stickyOutlineElement, providerID } = StickyElement.fromOutlineModel(outlineModel, this._model?.outlineProviderId);
+			this._model = new StickyModel(textModel.uri, modelVersionId, stickyOutlineElement, providerID);
+			return true;
+		} else {
+			this._model = undefined;
+			return false;
+		}
+	}
 
+	private async stickyModelFromSyntaxFoldingProvider(textModel: ITextModel, modelVersionId: number, token: CancellationToken): Promise<boolean | undefined> {
+		const selectedProviders = FoldingController.getFoldingRangeProviders(this._languageFeaturesService, textModel);
+		if (selectedProviders.length > 0) {
+			const provider = new SyntaxRangeProvider(textModel, selectedProviders, () => this.stickyModelFromSyntaxFoldingProvider(textModel, modelVersionId, token), this._foldingLimitReporter, undefined);
+			return this.stickyModelFromFoldingProvider(textModel, modelVersionId, token, provider);
 		} else {
-			const foldingController = FoldingController.get(this._editor);
-			const foldingModel = await foldingController?.getFoldingModel();
-			if (token.isCancellationRequested) {
-				return;
-			}
-			if (foldingModel && foldingModel.regions.length !== 0) {
-				const foldingElement = StickyOutlineElement.fromFoldingModel(foldingModel);
-				this._model = new StickyOutlineModel(model.uri, modelVersionId, foldingElement, undefined);
-			} else {
+			return false;
+		}
+
+	}
+
+	private async stickyModelFromIndentationProvider(textModel: ITextModel, modelVersionId: number, token: CancellationToken): Promise<boolean | undefined> {
+		const provider = new IndentRangeProvider(textModel, this._languageConfigurationService, this._foldingLimitReporter);
+		return this.stickyModelFromFoldingProvider(textModel, modelVersionId, token, provider);
+	}
+
+	private async stickyModelFromFoldingProvider(textModel: ITextModel, modelVersionId: number, token: CancellationToken, provider: RangeProvider): Promise<boolean | undefined> {
+
+		// It is possible to cancel the promise which calculates the folding
+		const foldingRegionPromise = this._foldingRegionPromise = createCancelablePromise(token => provider.compute(token));
+		return foldingRegionPromise.then(foldingRegions => {
+			if (foldingRegions === null || foldingRegions.length === 0) {
```

Comment:
foldingRegions.length === 0 means that the foling provider could do it's work, but found no ranges.
-> should not fallback to the next one